### PR TITLE
Some more platform dependent fixes

### PR DESCRIPTION
**Pull request by levinli303**
1. Move UI related code in celutil/winutil to celestia/win32
2. Add cctype include for toupper
3. Add some utils for apple platform and undef functions for portable builds

### CODE REVIEW

File: src/celutil/appleutils.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+#pragma once
+
+#include <string>
+
+std::string AppleHomeDirectory();
```
Comment (375gnu) [inline src/celutil/appleutils.h:16]: could you use fs::path here instead?

**Review by 375gnu — CHANGES_REQUESTED** [review]

File: src/celutil/utf8.cpp [hunk 2]
```diff
@@ -961,8 +961,8 @@ static size_t greekChunkLength(const std::string& str)
         npos = true;
     }
 
-    if (sp != 0 && isdigit(str[sp - 1]))
-        while(sp != 0 && isdigit(str[sp - 1])) sp--;
+    if (sp != 0 && str[sp - 1] >= 0 && isdigit(str[sp - 1]))
```
Comment (375gnu) [inline src/celutil/utf8.cpp:964]: you are doing something strange here.

**Review by 375gnu — CHANGES_REQUESTED** [review]

File: src/celutil/utf8.cpp (hunk 2 above)
Comment (levinli303) [reply src/celutil/utf8.cpp:964]: I ran into a crash here with MSVC (debug). passing a negative value to isdigit aborts the application.

Comment (375gnu) [reply src/celutil/utf8.cpp:964]: then the correct way it to cast to uchar:
```
if (sp != 0 && isdigit(static_cast<unsigned char>(str[sp - 1])))
    while(sp != 0 && isdigit(static_cast<unsigned char>(str[sp - 1]))) sp--;
```

**Review by 375gnu — APPROVED** [review]

**Pull request merged**